Grammar/python.gram

```diff
@@ -263,8 +263,13 @@ expression[expr_ty] (memo):
     | disjunction
     | lambdef
 
+expression_for_lambda[expr_ty] (memo):
+    | a=disjunction 'if' b=disjunction 'else' c=expression { _Py_IfExp(b, a, c, EXTRA) }
+    | disjunction
+    | lambdef
+
 lambdef[expr_ty]:
-    | 'lambda' a=[lambda_parameters] ':' b=expression { _Py_Lambda((a) ? a : CHECK(empty_arguments(p)), b, EXTRA) }
+    | 'lambda' a=[lambda_parameters] ':' b=expression_for_lambda { _Py_Lambda((a) ? a : CHECK(empty_arguments(p)), b, EXTRA) }
```

Comment:
Why is this needed?

Reply:
Sorry, that was me working on a different issue and forgot to stash. The issue is:

```
>>> (lambda: x:=2)
  File "<stdin>", line 1
SyntaxError: cannot use assignment expressions with lambda
>>> 
```